integration-tests/src/test/java/oracle/kubernetes/operator/JrfInOperatorTest.java

```diff
@@ -403,6 +412,263 @@ public void testAutoAndCustomSitConfigOverrides() throws Exception {
+      domain1Map.put("domainUID", "jrfrestart");
+      domain1Map.put("adminNodePort", 30705);
+      domain1Map.put("t3ChannelPort", 30025);
+      domain1Map.put("voyagerWebPort", 30309);
+      domain1Map.put("rcuSchemaPrefix", "jrfrestart");
+      domain1Map.put("initialManagedServerReplicas", 4);
+
+      // run RCU script to load db schema
+      DBUtils.runRCU(rcuPodName, domain1Map);
+
+      // create domain
+      logger.info("Creating Domain & verifying the domain creation");
+      domain1 = new JRFDomain(domain1Map);
+      domain1.verifyDomainCreated();
+
+      String originalYaml =
+          getUserProjectsDir() + "/weblogic-domains/" + domain1.getDomainUid() + "/domain.yaml";
+
+      // Rolling restart the cluster by setting restartVersion at the cluster level
+      // Modify the original domain yaml to include restartVersion in cluster level
+      DomainCRD crd = new DomainCRD(originalYaml);
+      Map<String, Object> clusterRestartVersion = new HashMap();
+      clusterRestartVersion.put("restartVersion", "clusterV1");
+      clusterRestartVersion.put("maxUnavailable", new Integer(2));
+      crd.addObjectNodeToCluster(domain1.getClusterName(), clusterRestartVersion);
+      String modYaml = crd.getYamlTree();
+      logger.info(modYaml);
+
+      // Write the modified yaml to a new file
+      String restartTmpDir = getResultDir() + "/restarttemp";
+      Files.createDirectories(Paths.get(restartTmpDir));
+      Path path = Paths.get(restartTmpDir, "restart.cluster.yaml");
+      logger.log(Level.INFO, "Path of the modified domain.yaml :{0}", path.toString());
+      Charset charset = StandardCharsets.UTF_8;
+      Files.write(path, modYaml.getBytes(charset));
+
+      // Apply the new yaml to update the domain crd
+      logger.log(Level.INFO, "kubectl apply -f {0}", path.toString());
+      ExecResult exec = TestUtils.exec("kubectl apply -f " + path.toString());
+      logger.info(exec.stdout());
+
+      int expectedMsPodsCount =
+          (Integer) domain1.getDomainMap().get("initialManagedServerReplicas");
+      // TODO: this verification will fail due to bug 29678557
+      logger.info("Verifying the number of not ready MS pods can not exceed maxUnavailable value");
+      verifyMSPodsNotReadyCountNotExceedMaxUnAvailable(domain1, expectedMsPodsCount, 2);
+
+      // TODO: this verification will fail due to bug 29720185
+      logger.info("Verifying the number of MS pods");
+      if (getMSPodsCount(domain1) != expectedMsPodsCount) {
+        throw new Exception(
+            "The number of MS pods is not right, expect: "
+                + expectedMsPodsCount
+                + ", got: "
+                + getMSPodsCount(domain1));
+      }
+    } finally {
+      if (domain1 != null && (JENKINS || testCompletedSuccessfully)) {
+        domain1.destroy();
+      }
+    }
+    logger.info("SUCCESS - " + testMethod);
+  }
+
+  /**
+   * This is the test case to cover bug 29684570.
```

Comment:
Please describe what the test is doing instead of referring to internal bugdb numbers